### PR TITLE
AO3-6564 Reduce fields where images are shown

### PR DESCRIPTION
**Pull request by weeklies**
# Pull Request Checklist

* [x] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [x] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [x] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [x] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [x] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6564

## Purpose

Strip images in the following:
- bookmarker notes
- pseud descriptions
- admin banners
- comments on news posts
- comment field on PAC and Support forms

## Testing Instructions

See Jira issue.

## Credit

weeklies (she/her)

### CODE REVIEW

File: app/views/layouts/_banner.html.erb [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
-      <%=raw sanitize_field(@admin_banner, :content) %>
-    </blockquote>
-    <% if current_user.nil? %>
-      <p class="submit">
-        <%= link_to "&times;".html_safe, current_path_with(hide_banner: true), :class => 'showme action', :title => ts("hide banner") %>
-      </p>
-    <% else %>
-      <%= form_tag end_banner_user_path(current_user), :method => :post, :remote => true do %>
+<%# BACK END this seems giant and messy and confusing, pls can we review?
+  # FRONT END yes let us rewrite this %>
```
Comment (ceithir) [inline app/views/layouts/_banner.html.erb:2]: We can likely get rid of this thirteen years old short discussion.

Also, :+1: for the condition clean-up.

File: features/comments_and_kudos/add_comment.feature [hunk 2]
```diff
@@ -180,6 +180,17 @@ Scenario: Set preference and receive comment notifications of your own comments
     And "commenter" should be emailed
     And 1 email should be delivered to "commenter"
 
+Scenario: Wprk comment displays images
```
Comment (ceithir) [inline features/comments_and_kudos/add_comment.feature:183]: Wprk -> Work

File: features/other_a/pseuds.feature [hunk 3]
```diff
@@ -118,6 +118,18 @@ Scenario: Manage pseuds - add, edit
     And I should see "I wanted to add another fancy name"
     And I should not see "My new name (editpseuds)"
 
+Scenario: Pseud descriptions do not display images
+
```
Comment (ceithir) [inline features/other_a/pseuds.feature:122]: Typo blank line?

File: app/views/comments/_single_comment.html.erb [hunk 4]
```diff
@@ -53,7 +53,13 @@
...
           <p class="notice"><%= ts("This comment has been hidden by an admin.") %></p>
         <% end %>
-        <blockquote class="userstuff"><%=raw sanitize_field(single_comment, :comment_content) %></blockquote>
+        <blockquote class="userstuff">
+          <% if single_comment.ultimate_parent.is_a?(AdminPost) %>
+            <%= raw strip_images(sanitize_field(single_comment, :comment_content)) %>
+          <% else %>
+            <%= raw sanitize_field(single_comment, :comment_content) %>
+          <% end %>
+        </blockquote>
```
[the remark this answers is not in the document]
Comment (weeklies) [reply app/views/comments/_single_comment.html.erb:62]: I am happy to do this if removing the 'separation' between the sanitization and the stripping of images is sensible?

[the remark this answers is not in the document]
Comment (sarken) [reply app/views/comments/_single_comment.html.erb:62]: I think it makes sense, sure! Stripping the image tags is definitely a form of sanitization.

File: features/other_a/pseuds.feature (hunk 3 above)
Comment (weeklies) [reply features/other_a/pseuds.feature:122]: It's consistent with the file.

**Review by ceithir — APPROVED** [review]

**Comment by sarken** [comment]
Just a heads up this has merge conflicts!

**Review by sarken — COMMENTED** [review]
Ohh, you know what just occurred to me: for news post comments, we should also be stripping the images from comment emails, both for the admin comment mailer and the user comment mailer. 

Would you be up for making that change?

**Comment by weeklies** [comment]
Sorry @sarken, I don't think I'll have the time for a while.

**Comment by sarken** [comment]
No worries! Someone might push to your branch or open up a second PR with that change, though, since we'd like this for the next regular release after we update Ruby and Rails.

**Review by sarken — APPROVED** [review]

**Pull request merged**